src/lib/snarky_js_bindings/test_module/simple-zkapp.js

```diff
@@ -37,94 +42,207 @@ class SimpleZkapp extends SmartContract {
-  let zkappCommandJson = transaction.toJSON();
-
-  // mina-signer part
-  let client = new Client({ network: "testnet" });
-  let feePayerAddress = client.derivePublicKey(feePayerKeyBase58);
-  let feePayer = {
-    feePayer: feePayerAddress,
-    fee: transactionFee,
-    nonce: feePayerNonce,
-  };
-  let zkappCommand = JSON.parse(zkappCommandJson);
-  let { data } = client.signTransaction(
-    { zkappCommand, feePayer },
-    feePayerKeyBase58
-  );
-  let ok = await verify(proof, verificationKey.data);
-  if (!ok) throw Error("verification failed");
+// a special account that is allowed to pull out half of the zkapp balance, once
+let privilegedKey = PrivateKey.random();
+let privilegedAddress = privilegedKey.toPublicKey();
+
+let zkappTargetBalance = 10_000_000_000;
+let initialBalance = zkappTargetBalance;
+let initialState = Field(1);
+
+console.log(
+  `simple-zkapp.js: Running with zkapp address ${zkappKey
+    .toPublicKey()
+    .toBase58()}, fee payer address ${feePayerAddress.toBase58()} and graphql uri ${graphql_uri}\n\n`
+);
+
+console.log(`simple-zkapp.js: Starting integration test\n`);
+
+let zkapp = new NotSoSimpleZkapp(zkappAddress);
+await NotSoSimpleZkapp.compile();
+
+console.log("deploying contract\n");
+let tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    AccountUpdate.fundNewAccount(feePayerAddress);
+
+    zkapp.deploy();
+  }
+);
+await tx.prove();
+await (await tx.sign([feePayerKey, zkappKey]).send()).wait(waitParams);
+
+await fetchAccount({ publicKey: zkappAddress });
+let zkappAccount = Mina.getAccount(zkappAddress);
+
+// we deployed the contract with an initial state of 1
+expectAssertEquals(zkappAccount.appState[0], Field(1));
+
+// the fresh zkapp account shouldn't have any funds
+expectAssertEquals(zkappAccount.balance, UInt64.from(0));
 
-  console.log(data.zkappCommand);
+console.log("deposit funds\n");
+tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    zkapp.deposit(UInt64.from(initialBalance));
+  }
+);
+await tx.prove();
+await (await tx.sign([feePayerKey]).send()).wait(waitParams);
+
+await fetchAccount({ publicKey: zkappAddress });
+zkappAccount = Mina.getAccount(zkappAddress);
+
+// we deposit 10_000_000_000 funds into the zkapp account
+expectAssertEquals(zkappAccount.balance, UInt64.from(initialBalance));
+
+console.log("update 1\n");
+tx = await Mina.transaction(
+  { sender: feePayerAddress, fee: 100_000_000 },
+  () => {
+    zkapp.update(Field(30));
+  }
+);
+await tx.prove();
+await (await tx.sign([feePayerKey]).send()).wait(waitParams);
+
+await fetchAccount({ publicKey: zkappAddress });
+zkappAccount = Mina.getAccount(zkappAddress);
+
+// no balance change expected
+expectAssertEquals(zkappAccount.balance, UInt64.from(initialBalance));
+
+// we updated the zkapp state to 31. x = x.add(y) ---- 31 = 1 + 30
+expectAssertEquals(zkappAccount.appState[0], Field(31));
```

Comment:
could you remove this round of `fetchAccount` plus assert equals checks? then this would also test that `Mina.transaction` correctly figures out the account fields (state precondition) by itself, if even they aren't currently in the cache. (the cumulative state update is asserted to be correct below, so it doesn't hurt to remove the check here)